session.go

```diff
@@ -50,19 +53,19 @@ type Session struct {
 	client      *spanner.Client
 	adminClient *adminapi.DatabaseAdminClient
 	clientOpts  []option.ClientOption
+	priority    pb.RequestOptions_Priority
+	tc          *transactionContext
+	tcMutex     sync.Mutex // Guard a critical section for transaction.
+}
 
-	// for read-write transaction
-	rwTxn *spanner.ReadWriteStmtBasedTransaction
-	// rwTxn is supposed to be accessed concurrently from transaction handling and heartbeat,
-	// so we will use mutex to guard a critical section.
-	rwTxnMutex    sync.Mutex
-	sendHeartbeat bool
-
-	// for read-only transaction
-	roTxn *spanner.ReadOnlyTransaction
```

Comment:
I refactored this to put transaction-scope things into the different struct so that we can manage a transaction easily in the session.